### PR TITLE
Swagger update batch default values

### LINKED ISSUE #232: Review all the endpoints for documentation
When I check swagger I want to see all the endpoints that are running.

Please review if swagger is documenting all the endpoints

### PR DESCRIPTION
**Pull request by ricardoaguiar**
Partially resolves #232

### CODE REVIEW

File: src/server/api/routes/batch-default-values.router.js
Comment (cecastosic) [inline src/server/api/routes/batch-default-values.router.js]: I think by default this should be empty, not current stage

File: src/server/api/routes/batch-default-values.router.js
Comment (cecastosic) [inline src/server/api/routes/batch-default-values.router.js]: I would move storage after harvest, the order will be as it is in DB: seeding, propagation, maturity, harvest, storage, delivery

**Review by cecastosic — CHANGES_REQUESTED** [review]

**Comment by cecastosic** [comment]
Great job 💪 
I have one more comment as a general comment. For every endpoint there is a description for the authentication part like this:
 *Authentication: super_admin admin user
But that's related to authorization and the authorization middleware, which we use for several endpoints, but not on all. I would suggest change to:
*Authentication: user
and for endpoints that use the authorization middleware
*Authorization: super_admin admin 
or 
*Authorization: super_admin 
depends on the endpoint.

File: src/server/api/routes/batch-default-values.router.js
Comment (ricardoaguiar) [reply src/server/api/routes/batch-default-values.router.js]: I removed the default altogether. It has no effect on the outcome. I will leave it out.

File: src/server/api/routes/batch-default-values.router.js
Comment (ricardoaguiar) [reply src/server/api/routes/batch-default-values.router.js]: Fixed.

**Comment by ricardoaguiar** [comment]
Do you mean to remove the `super_admin` and `admin `for the endpoints that are `user` authorized?

I have changed them to:
Authentication: `super_admin` `admin`
Authorization: `user`

Isn't a bit more clear than having just one level, i.e. either authentication or authorization?

Thanks

**Comment by cecastosic** [comment]
> Authentication: `super_admin` `admin`

Authorization is related to roles (super_admin, admin...), authentication is related to is the user logged in or not. This endpoint doesn't have anything with roles ie there is no usage of authorization middleware. IMO all endpoints that don't use authorization middleware shouldn't have authorization in the description, only authentication.

File: src/server/api/routes/users.router.js
Comment (zkwsk) [inline src/server/api/routes/users.router.js]: Why are the colons HTML encoded? Is that really necessary? Further up you use plain colons.

**Review by zkwsk — DISMISSED** [review]
Interesting discussion about Authentication/Authorization in the comments. I think at this point it is all up to you to decide how you want to document stuff like that, all  care about is that it is consistent.

But my suggestion would be something like this:

Endpoint A:
Authentication: true
Authorization: super-user, admin

Endpoint B:
Authentication: true
Authorization: super-user, admin, user

Endpoint C:
Authentication: false

To me it feels a little more in line with my understanding of authentication/authorization.

@ricardoaguiar I had a small comment about HTML encoded entities. Please check that you cannot just type regular colons, it's a tiny thing though, so I'm approving and then you can sort out how what exactly yo do about authentication annotations together :)

File: src/server/api/routes/users.router.js
Comment (ricardoaguiar) [reply src/server/api/routes/users.router.js]: I tried to use normal colons during the project and found out that using regular colons inside the text description, breaks the app. I guess YAML does not like that. 

Screen Shot 2020-05-10 at 08 07 07

I just did a quick look on SO and found this solution: use quotes around the text descriptions and escape the colons with :: or \
I will implement that and remove the encodings. Good observation.

https://stackoverflow.com/questions/32991517/how-to-escape-colons-and-other-special-characters-in-a-yaml-string

https://yaml.org/spec/1.2/spec.html#id2776092

Comment (ricardoaguiar) [reply src/server/api/routes/users.router.js]: @zkwsk 
I agree with your suggestions for Authentication and Authorization. It makes sense. 
As @cecastosic mentioned, we are not checking roles on this endpoint. But the fact that we are not checking for roles does not mean that it will work without the firebase token. I just tried and I got a 401 in return - not authorized.
That means all endpoints require authentication. The user, be either super_admin, admin or user, have to log in to generate a firebase token, which at that point can be used to access the endpoint documentation. 
I think we should implement your suggestion.

@cecastosic What do you think?

Comment (cecastosic) [reply src/server/api/routes/users.router.js]: Once again, token is related to authentication (not authorization), and ALL endpoints are secured by authentication middleware (ie firebase token). You can choose how you want to document auth part, I am fine with @zkwsk 's solution. IMO is better to put authorization description on the endpoints where it makes sense (add batch and delete batch) and not copy/paste 
Authorization: super_admin, admin, user
on almost all endpoints. 
🙂

**Review by cecastosic — DISMISSED** [review]

**Comment by ricardoaguiar** [comment]
Final revision. 
We are still missing a couple of endpoints that we may have to implement at some point. 
All we need for now is to approve and merge the PR.

Thanks.

**Review by ricardoaguiar — COMMENTED** [review]
Needs approval before merge.

**Pull request closed without merging**
